includes/class-sensei-posttypes.php

```diff
@@ -417,27 +417,30 @@ public function setup_sensei_message_post_type() {
 		if ( ! isset( Sensei()->settings->settings['messages_disable'] ) || ! Sensei()->settings->settings['messages_disable'] ) {
 
 			$args = array(
-				'labels'              => $this->create_post_type_labels( $this->labels['sensei_message']['singular'], $this->labels['sensei_message']['plural'], $this->labels['sensei_message']['menu'] ),
-				'public'              => true,
-				'publicly_queryable'  => true,
-				'show_ui'             => true,
-				'show_in_menu'        => 'admin.php?page=sensei',
-				'show_in_nav_menus'   => true,
-				'query_var'           => true,
-				'exclude_from_search' => true,
-				'rewrite'             => array(
+				'labels'                => $this->create_post_type_labels( $this->labels['sensei_message']['singular'], $this->labels['sensei_message']['plural'], $this->labels['sensei_message']['menu'] ),
+				'public'                => true,
+				'publicly_queryable'    => true,
+				'show_ui'               => true,
+				'show_in_menu'          => 'admin.php?page=sensei',
+				'show_in_nav_menus'     => true,
+				'query_var'             => true,
+				'exclude_from_search'   => true,
+				'rewrite'               => array(
 					'slug'       => esc_attr( apply_filters( 'sensei_messages_slug', _x( 'messages', 'post type single slug', 'sensei-lms' ) ) ),
 					'with_front' => false,
 					'feeds'      => false,
 					'pages'      => true,
 				),
-				'map_meta_cap'        => true,
-				'capability_type'     => 'question',
-				'has_archive'         => true,
-				'hierarchical'        => false,
-				'menu_position'       => 50,
-				'supports'            => array( 'title', 'editor', 'comments' ),
-				'delete_with_user'    => true,
+				'map_meta_cap'          => true,
+				'capability_type'       => 'question',
+				'has_archive'           => true,
+				'hierarchical'          => false,
+				'menu_position'         => 50,
+				'show_in_rest'          => true,
+				'rest_base'             => 'messages',
```

Comment:
Since we're putting this into the `wp/v2` namespace, we should probably make this more specific (as you mentioned in the PR description). Maybe `sensei-messages`?

Alternatively, I wonder whether we should put this in our own namespace, possible as a "private" endpoint? See discussion at p6rkRX-14Q-p2. I'm not currently convinced that this is necessary for this endpoint, though.

Reply:
Yeap we should make the endpoint more specific. I am not conviced that this should be private either.

As for naming I would say that the best approach would be to add a new endpoint i.e. /sensei/messages. Any sensei related endpoints should go under sensei.

Reply:
Just seen that we have many endpoints already which don't follow the sensei/<resource> convention. This means that sensei-messages is our only option.